app/views/RoomInfoEditView/index.tsx

```diff
@@ -1,268 +1,131 @@
-			description: '',
 			topic: '',
 			announcement: '',
-			joinCode: '',
-			nameError: {},
-			t: false,
-			ro: false,
-			reactWhenReadOnly: false,
-			archived: false,
-			systemMessages: [],
-			enableSysMes: false,
-			encrypted: false
-		};
-		this.loadRoom();
-	}
-
-	componentWillUnmount() {
-		if (this.querySubscription && this.querySubscription.unsubscribe) {
-			this.querySubscription.unsubscribe();
-		}
-	}
-
-	loadRoom = async () => {
-		const {
-			route,
-			setReadOnlyPermission,
-			setReactWhenReadOnlyPermission,
-			archiveRoomPermission,
-			unarchiveRoomPermission,
-			deleteCPermission,
-			deletePPermission,
-			deleteTeamPermission
-		} = this.props;
-		const rid = route.params?.rid;
-		if (!rid) {
-			return;
-		}
-		try {
-			const db = database.active;
-			const sub = await db.get('subscriptions').find(rid);
-			const observable = sub.observe();
-
-			this.querySubscription = observable.subscribe(data => {
-				this.room = data;
-				this.init(this.room);
-			});
-
-			const result = await hasPermission(
-				[
-					setReadOnlyPermission,
-					setReactWhenReadOnlyPermission,
-					archiveRoomPermission,
-					unarchiveRoomPermission,
-					deleteCPermission,
-					deletePPermission,
-					...(this.room.teamMain ? [deleteTeamPermission] : [])
-				],
-				rid
-			);
-
-			this.setState({
-				permissions: {
-					'set-readonly': result[0],
-					'set-react-when-readonly': result[1],
-					'archive-room': result[2],
-					'unarchive-room': result[3],
-					'delete-c': result[4],
-					'delete-p': result[5],
-					...(this.room.teamMain && { 'delete-team': result[6] })
-				}
-			});
-		} catch (e) {
-			log(e);
+			description: '',
+			joinCode: ''
 		}
-	};
+	});
 
-	init = (room: ISubscription) => {
+	const initializeRoomState = (room: ISubscription) => {
 		const { description, topic, announcement, t, ro, reactWhenReadOnly, joinCodeRequired, encrypted } = room;
 		const sysMes = room.sysMes as string[];
 		// fake password just to user knows about it
-		this.setState({
-			room,
-			name: getRoomTitle(room),
-			description,
-			topic,
-			announcement,
-			t: t !== 'p',
-			ro,
-			reactWhenReadOnly,
-			joinCode: joinCodeRequired ? this.randomValue : '',
-			archived: room.archived,
-			systemMessages: sysMes,
-			enableSysMes: sysMes && sysMes.length > 0,
-			encrypted
-		});
-	};
-
-	clearErrors = () => {
-		this.setState({
-			nameError: {}
-		});
-	};
-
-	reset = () => {
-		logEvent(events.RI_EDIT_RESET);
-		this.clearErrors();
-		this.init(this.room);
+		setValue('name', getRoomTitle(room));
+		setValue('description', description || '');
+		setValue('topic', topic || '');
+		setValue('announcement', announcement || '');
+		setValue('joinCode', joinCodeRequired ? RANDOM_VALUE : '');
+		setT(t !== 'p');
+		setReadOnly(ro);
+		setReactWhenReadOnly(reactWhenReadOnly);
+		setArchived(room.archived);
+		setSystemMessages(sysMes);
+		setEnableSysMes(sysMes && sysMes.length > 0);
+		setEncrypted(encrypted);
```

Comment:
Shouldn't we manage all form fields using the same technique on react-hook-form?